Clarify error & fallback handling (#879)
unicode-org/message-format-wg: spec/syntax.md

When the _matcher_ is processed, the result will be a single _pattern_ that serves
as the template for the formatting process.

A _message_ can only be considered _valid_ if the following requirements are satisfied;
otherwise, a corresponding _Data Model Error_ will be produced during processing:

- _Variant Key Mismatch_:
  The number of _keys_ on each _variant_ MUST be equal to the number of _selectors_.
- _Missing Fallback Variant_:
  At least one _variant_ MUST exist whose _keys_ are all equal to the "catch-all" key `*`.
- _Missing Selector Annotation_:
  Each _selector_ MUST have an _annotation_,

>   or contain a _variable_ that directly or indirectly references a _declaration_ with an _annotation_.

Reviewer: This text might change on Monday 😸

Author: I'm aware; will resolve conflicts after whichever lands first.

Multiple _attributes_ are permitted in an _expression_ or _markup_.
Each _attribute_ is separated by whitespace.

Each _attribute_'s _identifier_ SHOULD be unique within the _expression_ or _markup_:

> _attributes_ beyond the first with the same _identifier_ are ignored.

Author: ```suggestion
all but the last _attribute_ with the same _identifier_ are ignored.
```
